compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.desktop.kt

```diff
@@ -139,6 +139,36 @@ internal class ComposeContainer(
         composeSceneFactory = ::createComposeScene,
     )
 
+    /**
+     * The listener to [window] size and position changes.
+     */
+    private val windowComponentListener = object : ComponentAdapter() {
+        override fun componentResized(e: ComponentEvent?) = onChangeWindowSize()
+        override fun componentMoved(e: ComponentEvent?) = onChangeWindowPosition()
+    }
+
+    /**
+     * The listener to [_windowContainer] size and position changes.
+     */
+    private val windowContainerComponentListener = object : ComponentAdapter() {
+        override fun componentResized(e: ComponentEvent?)  {
+            if (!container.isDisplayable) return
+
+            mediator.onChangeComponentSize()
```

Comment:
It's still incorrent. `onChangeWindowSize` **must** be called here - we need to update the size here. Look, `windowContainer`'s size is used there

Reply:
Right, I'll move 
```
        windowContext.setContainerSize(windowContainer.sizeInPx)
```
here instead.

Reply:
Well, actually not move, but copy.

Reply:
this line not enough, layers must be updated too

Reply:
So the names of the functions on the layers and their javadoc are wrong?

Reply:
Window size reference for Compose is windowContainer, not real window. Compose shouldn't depend on "real" window for sizing (it does depend for things like window state/focus/etc)
We can rename window -> windowContainer in a few places to improve readability.

For the specific case, I'd avoid _removing_ re-checking size from callbacks and limit this PR only to _adding_ new callbacks

Reply:
I don't think that's correct for all cases. For example you would want to center dialogs in the window rather than the window container.

Reply:
That's exactly the case! window container IS the referece for that